code/game/machinery/vending/cm_vending.dm

```diff
@@ -446,6 +430,23 @@ IN_USE						used for vending/denying
 /obj/structure/machinery/cm_vending/proc/vend_succesfully()
 	return
 
+/obj/structure/machinery/cm_vending/proc/can_access(mob/user)
+	if(!hacked)
+		if(!allowed(user))
+			return list(FALSE, "Access denied.")
+
+		var/mob/living/carbon/human/H = user
+		var/obj/item/card/id/I = H.wear_id
+		if(!istype(I))
+			return list(FALSE, "Access denied. No ID card detected")
+
+		if(I.registered_name != user.real_name)
+			return list(FALSE, "Wrong ID card owner detected.")
+
+		if(LAZYLEN(vendor_role) && !vendor_role.Find(user.job))
+			return list(FALSE, "This machine isn't for you.")
+	return list(TRUE, "")
```

Comment:
I don't think these to_chats are showing

Reply:
I'm not sure why these weren't showing. Whilst I do prefer this style of function, I'll make the changes as I want to get moving onto something else.